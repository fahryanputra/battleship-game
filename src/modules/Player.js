import Gameboard from "./Gameboard";

class Player {
  constructor(name, size = 3) {
    this.name = name;
    this.gameboard = new Gameboard(size);
  }

  attack(player, x, y) {
    player.getGameboard().receiveAttack(x, y);
  }

  // Getter
  getName() {
    return this.name;
  }

  getGameboard() {
    return this.gameboard;
  }

  getLose() {
    return this.gameboard.getIsAllShipSunk();
  }
}

export default Player;
