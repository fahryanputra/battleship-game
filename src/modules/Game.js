import Player from "./Player";

class Game {
  constructor(players) {
    this.players = players;
    this.winner = "";
  }

  // Getter
  getPlayers() {
    return this.players;
  }

  getWinner() {
    return this.winner;
  }

  // Setter
  setPlayers(players) {
    this.players = players;
  }

  setWinner(winner) {
    this.winner = winner;
  }
}

export default Game;
