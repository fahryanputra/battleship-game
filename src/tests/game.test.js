import { describe, expect, test } from "@jest/globals";
import Game from "../modules/Game";
import Player from "../modules/Player";

const players = [];

const player = new Player("player", 3);
const computer = new Player("computer", 3);
players.push(player);
players.push(computer);

const game = new Game(players);

describe("Game class", () => {
  test("return player", () => {
    expect(game.getPlayers()[0]).toMatchSnapshot({
      name: "player",
    });
  });

  test("return computer", () => {
    expect(game.getPlayers()[1]).toMatchSnapshot({
      name: "computer",
    });
  });
});
