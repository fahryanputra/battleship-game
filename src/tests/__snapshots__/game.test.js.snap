// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game class return computer 1`] = `
{
  "gameboard": Gameboard {
    "attackMessage": "",
    "availableShips": [],
    "board": [
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
    ],
    "isAllShipSunk": false,
    "ships": [],
    "sunkMessage": "",
    "x": 3,
    "y": 3,
  },
  "name": "computer",
}
`;

exports[`Game class return player 1`] = `
{
  "gameboard": Gameboard {
    "attackMessage": "",
    "availableShips": [],
    "board": [
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
    ],
    "isAllShipSunk": false,
    "ships": [],
    "sunkMessage": "",
    "x": 3,
    "y": 3,
  },
  "name": "player",
}
`;

exports[`Game class return players 1`] = `
{
  "gameboard": Gameboard {
    "attackMessage": "",
    "availableShips": [],
    "board": [
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
      [
        0,
        0,
        0,
      ],
    ],
    "isAllShipSunk": false,
    "ships": [],
    "sunkMessage": "",
    "x": 3,
    "y": 3,
  },
  "name": "player",
}
`;
